### PR TITLE
Feature/hsi 298 implement p256 crypto key type

### PR DESCRIPTION
**Pull request by pbrito-curity**
Adds support for P256 JWS signer and verifier

### CODE REVIEW

File: JOSESwift/Sources/P256Verifier.swift [hunk 1]
```diff
@@ -1,9 +1,9 @@
 //
-//  SecureEnclaveVerifier.swift
+//  P256Verifier.swift
 //  JOSESwift
 //
 //  ---------------------------------------------------------------------------
-//  Copyright 2020 Curity AB
+//  Copyright 2024 Curity AB
```
Comment (renatoathaydes) [inline JOSESwift/Sources/P256Verifier.swift:6]: Revert this, we do not update year.

Comment (pbrito-curity) [reply JOSESwift/Sources/P256Verifier.swift:6]: It's a new file but git is showing it is as an update... there was a previous SecureEnclaveVerifier which was replaced with P256Verifier

File: JOSESwift/Sources/P256Verifier.swift
Comment (hlacure) [inline JOSESwift/Sources/P256Verifier.swift:1]: There are 2 SwiftLint warnings: trailing_whitespace
line 28 and 31

File: JOSESwift/Sources/P256Signer.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+//  ---------------------------------------------------------------------------
+//
+
+import CryptoKit
+import Foundation
+
+@available(iOS 13.0, *)
+internal struct P256Signer: SignerProtocol {
+    typealias KeyType = P256.Signing.PrivateKey
+
```
Comment (hlacure) [inline JOSESwift/Sources/P256Signer.swift:28]: Swiftlint warning: trailing_whitespace

File: JOSESwift/Sources/Verifier.swift
Comment (hlacure) [inline JOSESwift/Sources/Verifier.swift:1]: Each key type has a signer and verifier. P256 Verifier/Signer HMAC Verifier/Signer
Do SecureEnclave and P256 share the same verifier? I was expecting to see SecureEnclaveVerfier being used here. (I see it was replaced with P256Verfier)

**Review by hlacure — APPROVED** [review]

File: JOSESwift/Sources/Verifier.swift
Comment (pbrito-curity) [reply JOSESwift/Sources/Verifier.swift:1]: Both SecureEnclave keypair and P256 keypair have the same type of Public Key - `P256.Signing.PublicKey` so it is unnecessary to have two Verifiers which do the same

File: JOSESwift/Sources/P256Verifier.swift
Comment (pbrito-curity) [reply JOSESwift/Sources/P256Verifier.swift:1]: swiftlint run and fixed the tool's reported warnings

File: JOSESwift/Sources/P256Signer.swift (hunk 2 above)
Comment (pbrito-curity) [reply JOSESwift/Sources/P256Signer.swift:28]: swiftlint run and fixed the tool's reported warnings

**Review by phylexm — APPROVED** [review]

**Pull request merged**